src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -808,21 +910,21 @@ await CallWithRequestTracing(async () =>
                         throw new InvalidOperationException($"{nameof(snapshot.SnapshotComposition)} for the selected snapshot with name '{snapshot.Name}' must be 'key', found '{snapshot.SnapshotComposition}'.");
                     }
 
-                    settingsEnumerable = client.GetConfigurationSettingsForSnapshotAsync(
+                    IAsyncEnumerable<ConfigurationSetting> settingsEnumerable = client.GetConfigurationSettingsForSnapshotAsync(
```

Comment:
Do we reload snapshots if RegisterAll triggers a refresh?

Reply:
It doesn't right now but based on the definition of it refreshing all key-values loaded outside of `ConfigureRefresh`, it should. 

Seems like there should be a way for me to get rid of the `GetRefreshedKeyValueCollections` method (which as mentioned in another comment is an inaccurate name anyway) and just use `LoadSelected` which loads snapshots along with some additional logic in `RefreshAsync`.

Reply:
Not sure if I can remove it entirely actually, but I can change it to include snapshots for now